library/src/main/java/com/phillipcalvin/iconbutton/IconButton.java

```diff
@@ -65,34 +66,47 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
         textPaint.getTextBounds(text, 0, text.length(), bounds);
 
         int textWidth = bounds.width();
-        int contentWidth = drawableWidth + iconPadding + textWidth;
-
+        int factor = (drawablePosition == DrawablePositions.BOTH) ? 2 : 1;
+        int contentWidth = drawableWidth + iconPadding * factor + textWidth;
         int contentLeft = (int) ((getWidth() / 2.0) - (contentWidth / 2.0));
```

Comment:
yup `horizontalPadding` sounds more accurate. i actually don't exactly understand how this line works: `setCompoundDrawablePadding(-horizontalPadding + iconPadding);`

compound drawable padding (afaik) = space between the drawable and the text right? so why is this not just `iconPadding`. (it obviously doesn't work, but was curious to understand the specifics if you could explain it to me).